### PR TITLE
fix: Can't create new users via CLI if username is disabled

### LINKED ISSUE #1076: Bug: Can't create new users via CLI if username is disabled
### PHP Version

8.3.3

### CodeIgniter4 Version

4.4.6

### Shield Version

1.0.2

### Which operating systems have you tested for this bug?

macOS, Linux

### Which server did you use?

apache

### Database

MySQL 5.7, 10.5.22-MariaDB

### Did you customize Shield?

I customized a lot of things primarily i disabled the username requirement for registration.

### What happened?

When I try to create a new user via the CLI command `spark shield:user create` I get this error:
```
[ErrorException]
Undefined array key "username"
at VENDORPATH/codeigniter4/shield/src/Commands/User.php:245

Backtrace:
  1    VENDORPATH/codeigniter4/shield/src/Commands/User.php:245
       CodeIgniter\Debug\Exceptions()->errorHandler(2, 'Undefined array key "username"', '/PATH/vendor/codeigniter4/shield/src/Commands/User.php', 245)

  2    VENDORPATH/codeigniter4/shield/src/Commands/User.php:139
       CodeIgniter\Shield\Commands\User()->setValidationRules()

  3    SYSTEMPATH/CLI/Commands.php:65
       CodeIgniter\Shield\Commands\User()->run([...])

  4    SYSTEMPATH/CLI/Console.php:46
       CodeIgniter\CLI\Commands()->run('shield:user', [...])

  5    ROOTPATH/spark:102
       CodeIgniter\CLI\Console()->run()
```

### Steps to Reproduce

Create a custom registration validation with no username requirement.
Then try to create a new user via CLI command.

### Expected Output

Create a user without the need to supply a username.

### Anything else?

The validation rules for CLI creation get pulled from this: vendor/codeigniter4/shield/src/Commands/User.php:221:`$rules = $validationRules->getRegistrationRules();` but on line 245 it sets it like so 
```
$this->validationRules = [
    'username' => $rules['username'],
    'email'    => $rules['email'],
    'password' => $rules['password'],
];
```

### PR DESCRIPTION
**Pull request by kenjis**
**Description**
Fixes  #1076

**Checklist:**
- [x] Securely signed commits
- [x] Component(s) with PHPDoc blocks, only if necessary or adds value
- [ ] Unit testing, with >80% coverage
- [ ] User guide updated
- [x] Conforms to style guide

### CODE REVIEW

**Review by datamweb — CHANGES_REQUESTED** [review]
```console

[CodeIgniter\Database\Exceptions\DataException]

There is no data to insert.

at SYSTEMPATH\BaseModel.php:1703

Backtrace:
  1    SYSTEMPATH\BaseModel.php:1703
       CodeIgniter\Database\Exceptions\DataException::forEmptyDataset('insert')

  2    SYSTEMPATH\BaseModel.php:742
       CodeIgniter\BaseModel()->transformDataToArray([], 'insert')

  3    SYSTEMPATH\Model.php:730
       CodeIgniter\BaseModel()->insert(Object(CodeIgniter\Shield\Entities\User), false)

  4    VENDORPATH\codeigniter4\shield\src\Models\UserModel.php:265
       CodeIgniter\Model()->insert(Object(CodeIgniter\Shield\Entities\User), false)

  5    SYSTEMPATH\BaseModel.php:692
       CodeIgniter\Shield\Models\UserModel()->insert(Object(CodeIgniter\Shield\Entities\User), false)

  6    VENDORPATH\codeigniter4\shield\src\Models\UserModel.php:327
       CodeIgniter\BaseModel()->save(Object(CodeIgniter\Shield\Entities\User))

  7    VENDORPATH\codeigniter4\shield\src\Commands\User.php:304
       CodeIgniter\Shield\Models\UserModel()->save(Object(CodeIgniter\Shield\Entities\User))

  8    VENDORPATH\codeigniter4\shield\src\Commands\User.php:162
       CodeIgniter\Shield\Commands\User()->create(null, 'hghg@fff.it')

  9    SYSTEMPATH\CLI\Commands.php:65
       CodeIgniter\Shield\Commands\User()->run([...])

 10    SYSTEMPATH\CLI\Console.php:37
       CodeIgniter\CLI\Commands()->run('shield:user', [...])

 11    ROOTPATH\spark:97
       CodeIgniter\CLI\Console()->run()
```

File: src/Commands/User.php [hunk 1]
```diff
@@ -299,7 +302,12 @@ private function create(?string $username = null, ?string $email = null): void
...
         $user = new UserEntity($data);
-        $userModel->save($user);
+
+        if ($username === null) {
+            $userModel->allowEmptyInserts()->save($user);
+        } else {
+            $userModel->save($user);
+        }
 
         $this->write('User "' . $username . '" created', 'green');
```
Comment (datamweb) [inline src/Commands/User.php]: If there is no username, the appropriate message will not be printed. Please fix this issue.

Comment (kenjis) [reply src/Commands/User.php]: How?

Comment (datamweb) [reply src/Commands/User.php]: @kenjis Currently, the message output is as follows.
`User "" created`
Maybe message `New user created.` is enough.

Comment (sanchawebo) [reply src/Commands/User.php]:
```php
        if ($username === null) {
            $userModel->allowEmptyInserts()->save($user);
            $this->write('New User created', 'green');
        } else {
            $userModel->save($user);
            $this->write('User "' . $username . '" created', 'green');
        }
```

Comment (kenjis) [reply src/Commands/User.php]: Done.

**Review by datamweb — APPROVED** [review]

**Review by sanchawebo — APPROVED** [review]

**Pull request merged**